cmd/nerdctl/container_run_test.go

```diff
@@ -467,3 +468,21 @@ func TestRunWithTtyAndDetached(t *testing.T) {
 	withTtyContainer := base.InspectContainer(withTtyContainerName)
 	assert.Equal(base.T, 0, withTtyContainer.State.ExitCode)
 }
+
+func TestRunWithDetachKeys(t *testing.T) {
+	t.Parallel()
+
+	base := testutil.NewBase(t)
+	if testutil.GetTarget() == testutil.Nerdctl {
+		testutil.RequireDaemonVersion(base, ">= 1.6.0-0")
+	}
+
+	containerName := testutil.Identifier(t)
+	opts := []func(*testutil.Cmd){
+		testutil.WithStdin(bytes.NewReader([]byte{17, 16})), // https://www.physics.udel.edu/~watson/scen103/ascii.html
+	}
+	base.Cmd("run", "-it", "--detach-keys=ctrl-q,ctrl-p", "--name", containerName, testutil.CommonImage).CmdOption(opts...).AssertOK()
+	defer base.Cmd("container", "rm", "-f", containerName).AssertOK()
```

Comment:
please put defer before this [line](https://github.com/containerd/nerdctl/pull/2128/files#diff-2e856777b9ddfcd376cce19411cd5372a666ea2c5a6c3f4adeda04c983a14f5eR484)

Reply:
fixed